File: examples/python/end_user/revoke_delegation.py
# Usage: uv run python end_user/revoke_delegation.py <user_id>
# Note: This example requires the end user to have an active delegation on their account
# that allows the developer to sign on their behalf.

import asyncio
import sys

from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    if len(sys.argv) < 2:
        print("Usage: uv run python end_user/revoke_delegation.py <user_id>")
        sys.exit(1)

    user_id = sys.argv[1]

    async with CdpClient() as cdp:
        try:
            # Revoke delegation using the client method.
            await cdp.end_user.revoke_delegation(user_id=user_id)

            print("Revoked delegation (client method) for user:", user_id)

            # Alternatively, revoke delegation using the account method.
            # end_user = await cdp.end_user.get_end_user(user_id=user_id)

            # await end_user.revoke_delegation()

            # print("Revoked delegation (account method) for user:", user_id)

        except Exception as e:
            print(f"Error revoking delegation: {e}")
            raise e


asyncio.run(main())

File: examples/java/src/main/java/com/coinbase/cdp/examples/enduser/RevokeDelegation.java
package com.coinbase.cdp.examples.enduser;

import com.coinbase.cdp.CdpClient;
import com.coinbase.cdp.examples.utils.EnvLoader;

/**
 * Example: Revoke all active delegations for an end user.
 *
 * <p>This example demonstrates how to revoke all active delegations for a specified end user. After
 * revocation, the developer will no longer be able to perform delegated signing or sending
 * operations on behalf of the end user.
 *
 * <p>Usage: ./gradlew runRevokeDelegation --args="&lt;userId&gt;"
 */
public class RevokeDelegation {

  public static void main(String[] args) throws Exception {
    if (args.length < 1) {
      System.err.println("Usage: RevokeDelegation <userId>");
      System.exit(1);
    }

    String userId = args[0];
    EnvLoader.load();

    try (CdpClient cdp = CdpClient.create()) {
      System.out.println("Revoking all delegations for user: " + userId);

      cdp.endUser().revokeDelegation(userId);

      System.out.println("All delegations revoked successfully!");
    }
  }
}
